csharp/src/Apache.Arrow/Table.cs

```diff
@@ -66,12 +67,21 @@ public Table()
         }
 
 
-        public void RemoveColumn(int columnIndex)
+        public void RemoveColumn(int columnIndex, out Table newTable)
```

Comment:
Let's return the `Table` instead of using an `out` parameter. Typical C# uses the return by default. `out` parameters are used only when there are more than 1 value to return.